lib/sinon/proxy.js

```diff
@@ -123,6 +123,7 @@ var proxyApi = {
         this.thirdCall = null;
         this.lastCall = null;
         this.args = [];
+        this.firstArg = null;
```

Comment:
It's a bit weird/inconsistent that the values start out as `undefined`, but become `null` after being `reset()`.

Not sure this needs to be fixed, but just making note of it.

Reply:
I noticed this too. I was going to change it, but didn't want to assume whether the values should be `undefined` or `null`. (My vote is `undefined`.)

Reply:
I agree with the idea of returning to `undefined`.  Do you want to make the change, before we merge this in?

Reply:
Sure. Did you just want me to change this line to `undefined`? I feel like it'd be weird to have `firstArg` and `lastArg` default to different values, but changing `lastArg`'s default value is a big change. Maybe we could do that in a separate PR and just change `firstArg` to default to `undefined` for now?

Reply:
I agree. Let's leave `lastArg` unchanged in this PR.

A separate PR to tidy up the initial and reset values would be most welcome, and will trigger a new major release.